lib/puppet/type/foreman_instance_host.rb

```diff
@@ -0,0 +1,49 @@
+Puppet::Type.newtype(:foreman_instance_host) do
+  desc 'foreman_instance_host marks a host as foreman.'
```

Comment:
The language here gets tricky. Borrowing your language from other PRs, maybe something like:

```
foreman_instance_host marks a host as belonging to the set of hosts that make up the Foreman instance/application
```